src/inspect_tool_support/src/inspect_tool_support/_cli/main.py

```diff
@@ -0,0 +1,91 @@
+import argparse
+import asyncio
+import subprocess
+import sys
+from typing import Literal
+
+from jsonrpcserver import async_dispatch
+from pydantic import BaseModel
+
+from inspect_tool_support import SERVER_PORT
+from inspect_tool_support._cli._post_install import post_install
+from inspect_tool_support._cli.server import main as server_main
+from inspect_tool_support._util._common_types import JSONRPCResponseJSON
+from inspect_tool_support._util._json_rpc_helpers import json_rpc_http_call
+from inspect_tool_support._util._load_tools import load_tools
+
+_SERVER_URL = f"http://localhost:{SERVER_PORT}/"
+
+
+class JSONRPCRequest(BaseModel):
+    jsonrpc: Literal["2.0"]
+    method: str
+    id: int | float | str
+    params: list[object] | dict[str, object] | None = None
+
+
+def main() -> None:
+    args = _parse_args()
+    match args.command:
+        case "exec":
+            asyncio.run(_exec())
+        case "post-install":
+            post_install(no_web_browser=args.no_web_browser)
+        case "server":
+            server_main()
+
+
+# Example/testing requests
+# {"jsonrpc": "2.0", "method": "editor", "id": 666, "params": {"command": "view", "path": "/tmp"}}
+# {"jsonrpc": "2.0", "method": "bash", "id": 666, "params": {"command": "ls ~/Downloads"}}
+async def _exec() -> None:
+    _ensure_server_is_running()
```

Comment:
Launching the server on demand here. Gets rid of the need to install any sort of daemon, which would be wicked hard given that we don't know what kind of container/os we're getting installed into.